lib/src/core/mod.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{Context, Result};
+extern crate serde;
+use self::serde::{Deserialize, Serialize};
+use crate::*;
+
+pub mod proxy;
+pub mod verifier;
+use verifier::Verifier;
+use verifier::*;
+
+#[derive(Debug)]
+pub enum File {
+    Policy,
+    Reference,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Evidence {
+    pub nonce: String,
+    pub tee: String,
+    #[serde(rename = "tee-pubkey")]
```

Comment:
I googled and didn't find some key word like `rename_pattern`.
Actually the current implement have already designated the the rename pattern explictly. As the following:
```
    #[serde(rename = "tee-pubkey")]
    pub tee_pubkey: String,
```
It means rename the input's `tee-pubkey` -> `tee_pubkey`

Reply:
Sure. I wasn't expecting you to do this, but I'm saying it would be great if `serde` allowed you to apply the pattern to the *type*, not the field. That way, you could specify a single pattern potentially, rather than one for every single field in the type.

Reply:
OK, But seems the [Serde](https://serde.rs/field-attrs.html) doesn't support this rename/replace approch.